hwilib/devices/coldcard.py

```diff
@@ -68,53 +69,72 @@ def _get_fingerprint_hex(self):
     def sign_tx(self, tx):
         self.device.check_mitm()
 
-        # Get psbt in hex and then make binary
-        fd = io.BytesIO(base64.b64decode(tx.serialize()))
-
-        # learn size (portable way)
-        sz = fd.seek(0, 2)
-        fd.seek(0)
-
-        left = sz
-        chk = sha256()
-        for pos in range(0, sz, MAX_BLK_LEN):
-            here = fd.read(min(MAX_BLK_LEN, left))
-            if not here:
+        # Get this devices master key fingerprint
+        xpub = self.device.send_recv(CCProtocolPacker.get_xpub('m/0\''), timeout=None)
+        master_fp = get_xpub_fingerprint(xpub)
+
+        # For multisigs, we may need to do multiple passes if we appear in an input multiple times
```

Comment:
oof, what does this look like for the user? multiple independent signing passes?

Reply:
also, this needs a test

Reply:
It is multiple independent signings. There already are tests for this, our multisig tests use 3 keys from the same device. This was needed to pass our test suite.

Reply:
ah i see github was confusing me with commit ordering, looks good